lib/index.js

```diff
@@ -343,3 +361,16 @@ function _updateSessionAuthenticationData(
   }
   return data;
 }
+
+function _toBase64NoPad(x) {
+  const b32 = Buffer.from(x);
+  const b16 = Buffer.alloc(16);
+  b32.copy(b16, 0, 0, 16);
+  return b16.toString('base64').split('=')[0];
+}
```

Comment:
You can just use the `start` and `end` values in [Buffer.toString](https://nodejs.org/docs/latest-v18.x/api/buffer.html#buftostringencoding-start-end) to get the first 16 bytes.

```
function _toBase64NoPad(x) {
  return Buffer.from(x).toString('base64', 0, 16).split('=')[0];
}
```

However, the digest call you make already [returns a Buffer](https://nodejs.org/docs/latest-v18.x/api/crypto.html#hmacdigestencoding), so this could be simplified to just 
```
function _toBase64NoPad(x) {
  return x.toString('base64', 0, 16).split('=')[0];
}
```

It may not even need to be in a different function, you could just inline it above.

Reply:
Yes, `base64url` has been available since node 14:

https://nodejs.org/api/buffer.html#buffers-and-character-encodings